### PR TITLE
Improve boskos documentation, add a boskos config update job

### PR DESCRIPTION
**Pull request by krzyzacy**

### CODE REVIEW

**Comment by krzyzacy** [comment]
/assign @spxtr @cjwagner 

Woo the size plugin is working! Celebrate with a terrible joke:
/joke

**Comment by krzyzacy** [comment]
:-(((((((

/joke

**Comment by spxtr** [comment]
No joke for you!

**Comment by krzyzacy** [comment]
From prow log I can only see `msg:  "Issue comment created." ` and no actual error is logged?

**Comment by krzyzacy** [comment]
foo

/joke

**Comment by krzyzacy** [comment]
/joke

foo

**Comment by krzyzacy** [comment]
https://github.com/kubernetes/test-infra/blob/master/prow/plugins/yuks/yuks.go#L33 welp...

File: boskos/README.md [hunk 1]
```diff
@@ -134,6 +134,20 @@ On a successful request, `/metric` will return HTTP 200 and a JSON object contai
...
+[`Reaper`] will be look for inactive resource in free state but occupied (have not been updated for a long time) 
+actively, and reset the stale resources to dirty state for the [`Janitor`] component to pick up.
+
+[`Janitor`] looks for dirty resources from boskos, and will kick off sub-janitor process to clean up the 
+resource, finally return them back to boskos in a clean state.
+
+[`Metrics`] is a separate service, which can display json metric results, and has HTTP endpoint 
+opened for prometheus monitoring.
+
+For the boskos server handles k8s e2e jobs, the status is available from the [`Velodrome dashboard`]
```
Comment (cjwagner) [inline boskos/README.md]: s/handles/that handles/

File: boskos/README.md [hunk 2]
```diff
@@ -134,6 +134,20 @@ On a successful request, `/metric` will return HTTP 200 and a JSON object contai
 1. Boskos updates its config every 10min. Newly added resources will be available after next update cycle.
 Newly deleted resource will be removed in a future update cycle if the resource is not owned by any user.
 
+## Other Components:
+
+[`Reaper`] will be look for inactive resource in free state but occupied (have not been updated for a long time)
```
Comment (cjwagner) [inline boskos/README.md]: The first part of this sentence is confusing to me.

File: prow/config.yaml [hunk 3]
```diff
@@ -1260,6 +1260,32 @@ postsubmits:
         hostPath:
           path: /mnt/disks/ssd0
 
+  - name: maintenance-boskos-config-upload
+    agent: kubernetes
+    branches:
+    - master
```
Comment (cjwagner) [inline prow/config.yaml]: Does this make this post submit job run only when changes are merged to the master branch or does it do something else?

Comment (krzyzacy) [inline prow/config.yaml]: yeah (though we only have one branch in test-infra), we do the same thing for testgrid.

File: boskos/README.md (hunk 1 above)
Comment (krzyzacy) [inline boskos/README.md]: done

File: boskos/README.md (hunk 2 above)
Comment (krzyzacy) [inline boskos/README.md]: how about now :-)

File: jobs/config.json [hunk 4]
```diff
@@ -10079,6 +10079,15 @@
       "sig-testing"
     ]
   },
+  "maintenance-boskos-config-upload": {
+    "args": [
+      "cd boskos && make update-config"
```
Comment (krzyzacy) [inline jobs/config.json]:
```
Run: ('cd boskos && make update-config',)
OSError: [Errno 2] No such file or directory
```

¯\_(ツ)_/¯

**Pull request merged**